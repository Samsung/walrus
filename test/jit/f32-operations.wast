(module
(func (export "test1") (result f32)
    f32.const 3.3
    f32.const 4.4
    f32.max
)
(func (export "test2") (result f32)
    f32.const 3.3
    f32.const 4.4
    f32.min
)
(func (export "test3") (result f32)
    f32.const 4.0
    f32.sqrt
)
(func (export "test4") (result f32)
    f32.const 3.3
    f32.ceil
)
(func (export "test5") (result f32)
    f32.const 3.3
    f32.floor
)
(func (export "test6") (result f32)
    f32.const 3.3
    f32.trunc
)
(func (export "test7") (result f32)
    f32.const 3.3
    f32.nearest
)
(func (export "test8") (result f32)
    f32.const 3.3
    f32.abs
)
(func (export "test9") (result f32)
    f32.const -3.3
    f32.abs
)
(func (export "test10") (result f32)
    f32.const -3.3
    f32.neg
)

(func (export "test11") (result i32 i32 i32 i32 i32 i32)
    f32.const 3.3
    f32.const 3.3
    f32.eq

    if (result i32)
        i32.const 69
    else
        i32.const 1
    end

    f32.const 3.3
    f32.const -3.3
    f32.ne

    if (result i32)
        i32.const 68
    else
        i32.const 2
    end

    f32.const 3.3
    f32.const 3.0
    f32.lt

    if (result i32)
        i32.const 3
    else
        i32.const 67
    end

    f32.const 4.0
    f32.const 4.3
    f32.gt

    if (result i32)
        i32.const 4
    else
        i32.const 66
    end

    f32.const 3.3
    f32.const 4.4
    f32.le

    if (result i32)
        i32.const 65
    else
        i32.const 5
    end

    f32.const 4.4
    f32.const 3.3
    f32.ge

    if (result i32)
        i32.const 64
    else
        i32.const 6
    end
)

(func (export "test12") (result i32)
    f32.const 3.3
    f32.const 3.3
    f32.eq
)

(func (export "test13") (result i32)
    f32.const 3.3
    f32.const 3.3
    f32.ne
)

(func (export "test14") (result i32 i32 i32)
    f32.const nan:0x200000
    f32.const inf
    f32.ge
    f32.const -inf
    f32.const -inf
    f32.ge
    f32.const -inf
    f32.const -0x0p+0
    f32.eq
)

(func (export "test15") (result f32) (local f32 f32)
    f32.const 3.3
    local.set 0
    f32.const 4.4
    local.set 1

    f32.const 5.5
    f32.const 6.6
    local.get 0
    local.get 1
    f32.eq
    select (result f32)
)

(func (export "test16") (result i32) (local f32 f32)
    f32.const 3.3
    local.set 0
    f32.const 4.4
    local.set 1

    i32.const 5
    i32.const 6
    local.get 0
    local.get 1
    f32.eq

    select
)

(func (export "test17") (result i32 i32 i32 i32)
    f32.const -0x0p+0
    f32.const -nan
    f32.ne

    f32.const -0x0p+0
    f32.const -nan:0x200000
    f32.ne

    f32.const -0x1p-149
    f32.const -0x1p-149
    f32.ne

    f32.const 0x0p+0
    f32.const nan
    f32.ne
)
)

(assert_return (invoke "test1") (f32.const 4.4))
(assert_return (invoke "test2") (f32.const 3.3))
(assert_return (invoke "test3") (f32.const 2.0))
(assert_return (invoke "test4") (f32.const 4.0))
(assert_return (invoke "test5") (f32.const 3.0))
(assert_return (invoke "test6") (f32.const 3.0))
(assert_return (invoke "test7") (f32.const 3.0))
(assert_return (invoke "test8") (f32.const 3.3))
(assert_return (invoke "test9") (f32.const 3.3))
(assert_return (invoke "test10") (f32.const 3.3))
(assert_return (invoke "test11") (i32.const 69) (i32.const 68) (i32.const 67) (i32.const 66) (i32.const 65) (i32.const 64))
(assert_return (invoke "test12") (i32.const 1))
(assert_return (invoke "test13") (i32.const 0))
(assert_return (invoke "test14") (i32.const 0) (i32.const 1) (i32.const 0))
(assert_return (invoke "test15") (f32.const 6.6))
(assert_return (invoke "test16") (i32.const 6))
(assert_return (invoke "test17") (i32.const 1) (i32.const 1) (i32.const 0) (i32.const 1))
