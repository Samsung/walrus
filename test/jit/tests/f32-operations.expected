[[[[[[[  Function   0  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.const
  Result[0]: f32
2: Opcode: f32.max
  Param[0]: ref:0(0)
  Param[1]: ref:1(0)
  Result[0]: f32
3: Opcode: return
  Param[0]: ref:2(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.const
  Result[0]: f32 unused
2: Opcode: f32.max
  Param[0]: f32 const:0
  Param[1]: f32 const:1
  Result[0]: f32 sp[0]
3: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   1  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.const
  Result[0]: f32
2: Opcode: f32.min
  Param[0]: ref:0(0)
  Param[1]: ref:1(0)
  Result[0]: f32
3: Opcode: return
  Param[0]: ref:2(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.const
  Result[0]: f32 unused
2: Opcode: f32.min
  Param[0]: f32 const:0
  Param[1]: f32 const:1
  Result[0]: f32 sp[0]
3: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   2  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.sqrt
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.sqrt
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   3  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.ceil
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.ceil
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   4  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.floor
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.floor
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   5  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.trunc
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.trunc
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   6  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.nearest
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.nearest
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   7  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.abs
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.abs
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   8  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.abs
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.abs
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

[[[[[[[  Function   9  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f32
1: Opcode: f32.neg
  Param[0]: ref:0(0)
  Result[0]: f32
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f32 unused
1: Opcode: f32.neg
  Param[0]: f32 const:0
  Result[0]: f32 sp[0]
2: Opcode: return
  Param[0]: f32 sp[0]

test1() => f32:4.400000
test2() => f32:3.300000
test3() => f32:2.000000
test4() => f32:4.000000
test5() => f32:3.000000
test6() => f32:3.000000
test7() => f32:3.000000
test8() => f32:3.300000
test9() => f32:3.300000
test10() => f32:3.300000
